src/adapter/src/catalog/migrate.rs

```diff
@@ -152,6 +181,163 @@ fn csr_url_path_rewrite(stmt: &mut mz_sql::ast::Statement<Raw>) {
     }
 }
 
+// Add the `col_num` to all PG column descriptions.
+// TODO: delete in version v0.48 (released in v0.46 + 1 additional release)
+async fn pg_source_col_num_rewrite(
+    catalog: &ConnCatalog<'_>,
+    connection_context: &ConnectionContext,
+    stmt: &mut mz_sql::ast::Statement<Raw>,
+) {
+    use mz_sql::plan::StatementContext;
+    use mz_sql_parser::ast::{CreateSourceConnection, PgConfigOption};
+
+    if let Statement::CreateSource(mz_sql::ast::CreateSourceStatement {
+        name,
+        connection:
+            CreateSourceConnection::Postgres {
+                connection,
+                options,
+            },
+        ..
+    }) = stmt
+    {
+        // Find the option containing the serialized details.
+        let details_idx = options
+            .iter()
+            .position(|PgConfigOption { name, .. }| name == &PgConfigOptionName::Details)
+            .expect("corrupt catalog");
+
+        // Examine the current details
+        let details = match &options[details_idx].value {
+            Some(mz_sql::ast::WithOptionValue::Value(mz_sql::ast::Value::String(details))) => {
+                details
+            }
+            _ => unreachable!("corrupt catalog"),
+        };
+
+        let details = hex::decode(details).expect("valid catalog");
+        let details =
+            ProtoPostgresSourcePublicationDetails::decode(&*details).expect("valid catalog");
+        let mut publication_details =
+            PostgresSourcePublicationDetails::from_proto(details).expect("valid catalog");
+
+        if publication_details
+            .tables
+            .iter()
+            .all(|t| t.columns.iter().all(|c| c.col_num != 0))
```

Comment:
Is it 0 here because 0 is the default value? I think in general we're trying to add an `Option` to new fields and then remove the `Option` in the next release.

Reply:
Sure thing--did this.